src/transformers/trainer.py

```diff
@@ -896,7 +896,7 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
                     self.lr_scheduler.step()
                     model.zero_grad()
                     self.state.global_step += 1
-                    self.state.epoch = epoch + (step + 1) / steps_in_epoch
+                    self.state.epoch = round(epoch + (step + 1) / steps_in_epoch, 2)
```

Comment:
No the value should only be rounded when logged. Internal code could use `self.state.epoch` and needs the exact value.

Reply:
You are correct, logging time it is. Done